dfget/core/core.go

```diff
@@ -189,8 +190,36 @@ func downloadFile(cfg *config.Config, supernodeAPI api.SupernodeAPI,
 	reportFinishedTask(cfg, getter)
 
 	os.Remove(cfg.RV.TempTarget)
-	logrus.Infof("download %s cost:%.3fs length:%d reason:%d",
-		success, time.Since(cfg.StartTime).Seconds(), cfg.RV.FileLength, cfg.BackSourceReason)
+	downloadTime := time.Since(cfg.StartTime).Seconds()
+	if cfg.UploadMetrics {
+		req := &types.TaskMetricsRequest{
+			BacksourceReason: strconv.Itoa(cfg.BackSourceReason),
+			IP:               cfg.RV.LocalIP,
+			CID:              cfg.RV.Cid,
+			CallSystem:       cfg.CallSystem,
+			Duration:         downloadTime,
+			FileLength:       cfg.RV.FileLength,
+			Port:             int32(cfg.RV.PeerPort),
+			Success:          success,
+			TaskID:           result.TaskID,
+		}
+		for _, node := range cfg.Node {
+			resp, err := supernodeAPI.ReportMetrics(node, req)
+			if err != nil {
+				logrus.Error(err)
```

Comment:
Please see the error structure of function `ReportMetrics`, and you will find that this error log is unreadable when it is streamed in the logfile. please use:
```
logrus.Errorf("failed to report metrics from node %s to supernode node %s: %v", a, b, c)
```
WDYT?
Error and log construction is some kind of art. 😄

Reply:
Agree, I will update it